File: libs/typescript/core/src/http.ts
/**
 * Shared HTTP utilities for CUA SDK requests.
 */

export const CUA_CLIENT_VERSION_HEADER = 'X-Cua-Client-Version';

/**
 * Build a version header dict for CUA SDK requests.
 *
 * @param packageName - short package name, e.g. `'agent'`, `'computer'`
 * @param version     - semver string injected at build time
 * @returns header record; empty when version is falsy, so always safe to spread.
 */
export function cuaVersionHeaders(
  packageName: string,
  version: string,
): Record<string, string> {
  if (!version) return {};
  return { [CUA_CLIENT_VERSION_HEADER]: `${packageName}:${version}` };
}

File: libs/python/core/cua_core/http.py
"""Shared HTTP utilities for CUA SDK requests."""

from functools import lru_cache

CUA_CLIENT_VERSION_HEADER = "X-Cua-Client-Version"

# CUA packages whose versions are included in the header value.
_CUA_PACKAGES = ("cua-agent", "cua-computer", "cua-core")


@lru_cache(maxsize=1)
def _build_version_string() -> str:
    """Return a composite version string like ``agent:0.4.0 computer:0.1.0 core:0.1.8``."""
    from importlib.metadata import PackageNotFoundError, version

    parts: list[str] = []
    for pkg in _CUA_PACKAGES:
        try:
            short = pkg.removeprefix("cua-")
            parts.append(f"{short}:{version(pkg)}")
        except PackageNotFoundError:
            continue
    return " ".join(parts)


def cua_version_headers() -> dict[str, str]:
    """Return headers dict containing the CUA client version header.

    Only installed CUA packages are included.  If none are found the dict is
    empty so it is always safe to unpack with ``**cua_version_headers()``.
    """
    value = _build_version_string()
    if not value:
        return {}
    return {CUA_CLIENT_VERSION_HEADER: value}
